plugins/incremental-ingestion-backend/README.md

```diff
@@ -0,0 +1,328 @@
+
+1. The cursor must be serializable to JSON (not an issue for most RESTful or GraphQL based APIs).
+2. The client must be stateless - a client is created from scratch for each iteration to allow distributing processing over multiple replicas.
+3. There must be sufficient storage in Postgres to handle the additional data.
+
+## Installation
+
+1. Install `@backstage/plugin-incremental-ingestion-backend` with `yarn add @backstage/plugin-incremental-ingestion-backend`
+2. Import `IncrementalCatalogBuilder` from `@backstage/plugin-incremental-ingestion-backend` and instantiate it with `await IncrementalCatalogBuilder.create(env, builder)`. You have to pass `builder` into `IncrementalCatalogBuilder.create` function because `IncrementalCatalogBuilder` will convert an `IncrementalEntityProvider` into an `EntityProvider` and call `builder.addEntityProvider`.
+  ```ts
+  const builder = CatalogBuilder.create(env);
+  // incremental builder receives builder because it'll register
+  // incremental entity providers with the builder 
+  const incrementalBuilder = await IncrementalCatalogBuilder.create(env, builder);
+  ```
+3. Last step, add `await incrementBuilder.build()` after `await builder.build()` to ensure that all `CatalogBuider` migration run before running `incrementBuilder.build()` migrations.
+  ```ts
+  const { processingEngine, router } = await builder.build();
+
+  // this has to run after `await builder.build()` to ensure that catalog migrations are completed 
+  // before incremental builder migrations are executed 
+  await incrementalBuilder.build();
+  ```
+
+The result should look something like this,
+
+```ts
+import {
+  CatalogBuilder
+} from '@backstage/plugin-catalog-backend';
+import { ScaffolderEntitiesProcessor } from '@backstage/plugin-scaffolder-backend';
+import { IncrementalCatalogBuilder } from '@frontside/backstage-plugin-incremental-ingestion-backend';
+import { GithubRepositoryEntityProvider } from '@frontside/backstage-plugin-incremental-ingestion-github';
+import { Router } from 'express';
+import { Duration } from 'luxon';
+import { PluginEnvironment } from '../types';
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  
+  const builder = CatalogBuilder.create(env);
+  // incremental builder receives builder because it'll register
+  // incremental entity providers with the builder 
+  const incrementalBuilder = await IncrementalCatalogBuilder.create(env, builder);
+
+  builder.addProcessor(new ScaffolderEntitiesProcessor());
+
+  const { processingEngine, router } = await builder.build();
+
+  // this has to run after `await builder.build()` so ensure that catalog migrations are completed 
+  // before incremental builder migrations are executed 
+  await incrementalBuilder.build();
+  // If you want to use a suite of administrative routes to control the incremental providers, use
+  // `const { incrementalAdminRouter } = await incrementalBuilder.build();`
```

Comment:
We should just add this by default

Reply:
Done! It's best if the default catalog router uses the incremental admin router anyway.